File: ARCADE/Intro/Level 10_5: isMAC48Address.cs
    using System;
     
    public class Test
    {
    	public static bool isMAC48Address(string inputString) {
    	    for (int i = 0; i < inputString.Length; i++) {
    	        if(i % 3 == 2){
    	            if (inputString[i] != '-') return false;
    	        }
    	        else{
    	            char charr = inputString[i];
    	            if(!('0' <= charr && charr <= '9' || 'A' <= charr && charr <= 'F')) return false;
    	        }
    	    }
    	    if(inputString.Length != 17) return false;
    	    return true;
    	}
     
    	public static void Main()
    	{
    		string inputString = "00-1B-63-84-45-E6";
    		Console.WriteLine(isMAC48Address(inputString));
    		string inputString1 = "Z1-1B-63-84-45-E6";
    		Console.WriteLine(isMAC48Address(inputString1));
    	}
    }

File: ARCADE/Intro/Level 10_5: isMAC48Address.pas
program isMAC48Address;
 
function isMAC48Address(inputString: string): boolean;
var i : integer;
    rs : boolean;
begin
    if(length(inputString) <> 17) then
    begin
        isMAC48Address := false;
        exit;
    end;
    rs := true;
    for i := 1 to length(inputString) do
    begin
        if(i mod 3 = 0) then
        begin
            if(inputString[i] <> '-') then rs := false;
        end
        else
        begin
            if(not(((48 <= ord(inputString[i])) and (ord(inputString[i]) <= 57)) or 
            ((65 <= ord(inputString[i])) and (ord(inputString[i]) <= 70)))) then rs := false;
        end;
    end;
 
    isMAC48Address := rs;
end;
 
var inputString: string;
begin
	inputString := '00-1B-63-84-45-E6';
	writeln(isMAC48Address(inputString));
	inputString := 'Z1-1B-63-84-45-E6';
	writeln(isMAC48Address(inputString));
	inputString := 'not a MAC-48 address';
	writeln(isMAC48Address(inputString));
end.
